pandas/_libs/algos.pyx

```diff
@@ -1158,34 +1157,14 @@ def rank_2d(
 
     n, k = (<object>values).shape
     ranks = np.empty((n, k), dtype='f8')
-    # For compatibility when calling rank_1d
-    labels = np.zeros(k, dtype=np.int64)
 
-    if rank_t is object:
-        try:
-            _as = values.argsort(1)
-        except TypeError:
-            values = in_arr
-            for i in range(len(values)):
-                ranks[i] = rank_1d(
-                    in_arr[i],
-                    labels=labels,
-                    ties_method=ties_method,
-                    ascending=ascending,
-                    pct=pct
-                )
-            if axis == 0:
-                return ranks.T
-            else:
-                return ranks
+    if tiebreak == TIEBREAK_FIRST:
+        # need to use a stable sort here
+        _as = values.argsort(axis=1, kind='mergesort')
```

Comment:
can you rename _as -> lexsort_indexer

Reply:
Renamed to `argsort_indexer` (also have another cleanup branch with followups from #38744 to unify some other naming stuff, but ran into this issue on the way). The `lexsort`/`argsort` distinction is important here because `rank_1d` uses `lexsort` to sort by `(values, mask)`, but `rank_2d` just sorts by `values`. Will open an issue about this later if I can confirm this causes wrong behavior.